scripts/quests/windurst/Class_Reunion.lua

```diff
@@ -0,0 +1,185 @@
+
+                onEventFinish =
+                {
+                    [412] = function(player, csid, option, npc)
+                        player:delKeyItem(player, xi.ki.CARBUNCLES_TEAR)
+                        player:messageSpecial(ID.text.LOST_KEYITEM, xi.ki.CARBUNCLES_TEAR)
+                        quest:setVar(player, 'Prog', 2)
+                    end,
+
+                    [407] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 3)
+                    end,
+
+                    [410] = function(player, csid, option, npc)
+                        quest:complete(player)
+                    end,
+                },
+            },
+        },
+
+            ['Shantotto'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(409)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Fuepepe'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(817)
+                    end
+                end,
+            },
+
+            ['Furakku-Norakku'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(816)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Gulmama'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:progressEvent(713)
+                    elseif questProgress == 4 and not player:hasItem(xi.items.ICE_PENDULUM) then
+                        return quest:event(712)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [713] = function(player, csid, option, npc)
+                    npcUtil.giveItem(player, xi.items.ICE_PENDULUM)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [712] = function(player, csid, option, npc)
+                    npcUtil.giveItem(player, xi.items.ICE_PENDULUM)
+                end
+            },
+        },
+
+        [xi.zone.CLOISTER_OF_FROST] =
+        {
+            onEventFinish =
+            {
+                [32001] = function(player, csid, option, npc)
+                    if
+                        player:getLocalVar('battlefieldWin') == 577 and
```

Comment:
if players go and complete another battlefield where this variable is also set, it will overwrite player progress. I also don't see this variable being set anywhere.